### PR TITLE
Spiffs Magic2 & Mount Restructure

### PR DESCRIPTION
**Pull request by hreintke**

### CODE REVIEW

File: samples/Basic_rBoot/app/application.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ void ShowInfo() {
 void serialCallBack(Stream& stream, char arrivedChar, unsigned short availableCharsCount) {
 	
 
-	if (arrivedChar == '\n') {
+	if (arrivedChar == '\r') {
```
Comment (raburton) [inline samples/Basic_rBoot/app/application.cpp:103]: This is wrong!

Comment (hreintke) [inline samples/Basic_rBoot/app/application.cpp:103]: Sorry will revert now

**Comment by hreintke** [comment]
@raburton :
This PR updates the spiffs integration to the latest master of spiffs repo. Not yet a released Spiffs version but Peter Andersson committed to keep them in Spiffs.

New in this repo is : 
- A Magic_length config option which tests for correct filesystem when mounting a FS.
  Difference with Magic is the check is more secure and also validates FS size.
- A spiffs_probe function which returns the size of a FS, based on the start address.

By using the above functions I can circumvent the issue of not knowing the size of the spiffy generated FS'es and opening the FS with spiffs_mount(). 

Now there is : 
- spiffs_mount() - > mounts FS which is located after application, sized according to probe.
- spiffs_mount_manual(address, size) . use these for mounting
- spiffs_mount_manual(address, 0) use FS on address, sized according to probe.

This keeps backward compatibility.

Update : There is no backward compatibility according Filesystems.

**Comment by raburton** [comment]
I like this, good work!

**Comment by hreintke** [comment]
Thanks,
Just a little work on format and unmount and we have a clean and easy maintainable spiffs integration.

**Pull request merged**